srml/contract/src/gas.rs

```diff
@@ -213,18 +212,12 @@ pub fn buy_gas<T: Trait>(
 
 	// Buy the specified amount of gas.
 	let gas_price = <Module<T>>::gas_price();
-	let b = <balances::Module<T>>::free_balance(transactor);
 	let cost = <T::Gas as As<T::Balance>>::as_(gas_limit.clone())
 		.checked_mul(&gas_price)
 		.ok_or("overflow multiplying gas limit by price")?;
 
-	let new_balance = b.checked_sub(&cost);
-	if new_balance < Some(<balances::Module<T>>::existential_deposit()) {
-		return Err("not enough funds for transaction fee");
-	}
+	<T as Trait>::ChargeFee::charge_fee(transactor, cost)?;
```

Comment:
@pepyakin Maybe in general `ChargeFee` (and `TransferAsset` in the implementation) trait are not supposed to kill transactor's account. Account got killed is more like an unexpected side effect while paying fees or doing transfer in other srml modules except `balances`. I could update the implementation of `TransferAsset`, to make the functions return `Err`s instead of killing accounts.